cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -127,12 +127,24 @@ class parquet_field_int : public parquet_field {
 
   T& val;
 
+  std::string type_string()
```

Comment:
```suggestion
  std::string type_string() const
```

Reply:
It's better to use switch:
```
switch (EXPECTED_TYPE) {
 case ....: return ...;
}
```

Reply:
changed to a switch when I generalized the function so it can be reused